### PR TITLE
06-TimeManagment: Add kernel module for time

### PR DESCRIPTION
**Pull request by BekirovBekir**
Module allows to determine relative, absolute time and  average loading via sysfs

### CODE REVIEW

File: 06-TimeManagement/timers_module.c [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+						NS, &time_remainder));
+
+	ptr_timespec->tv_nsec = time_remainder;
+}
+
+static void relative_time_calc(struct timers_module_data *ptr,
+						 struct timespec *ptr_timespec)
+{
+	ptr->cur_jiffies = jiffies - ptr->prev_jiffies;
+	ptr->prev_jiffies = jiffies;
```
Comment (yekovalyov) [inline 06-TimeManagement/timers_module.c]: is the value of variable jiffies the same for these 2 lines?

File: 06-TimeManagement/timers_module.c [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+static void __exit timers_module_stop(void)
+{
+	class_remove_file(module_data->attr_class, &class_attr_relative_time);
+	class_remove_file(module_data->attr_class, &class_attr_absolute_time);
+	class_remove_file(module_data->attr_class, &class_attr_average_load);
+	class_destroy(module_data->attr_class);
+	del_timer(&module_data->timer);
+
+		if (module_data)
```
Comment (yekovalyov) [inline 06-TimeManagement/timers_module.c]: Did You check module using checkpatch.pl?

File: 06-TimeManagement/timers_module.c [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+	relative_time_calc(module_data, &relative_time);
+
+		if (module_data->first_rel_reading == 0) {
+			relative_time.tv_nsec = 0;
+			relative_time.tv_sec = 0;
+			module_data->first_rel_reading = 1;
+		}
+
+	sprintf(buf, "%lu.%lu s\n", relative_time.tv_sec,
```
Comment (yekovalyov) [inline 06-TimeManagement/timers_module.c]: is this format string correct for all possible values of tv_nsec?

File: 06-TimeManagement/timers_module.c [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+	ptr_timespec->tv_nsec = time_remainder;
+}
+
+static void relative_time_calc(struct timers_module_data *ptr,
+						 struct timespec *ptr_timespec)
+{
+	ptr->cur_jiffies = jiffies - ptr->prev_jiffies;
+	ptr->prev_jiffies = jiffies;
+	ptr_timespec->tv_sec = jiffies_to_msecs(ptr->cur_jiffies) / 1000;
+	ptr_timespec->tv_nsec = jiffies_to_msecs(ptr->cur_jiffies) % 1000;
```
Comment (yekovalyov) [inline 06-TimeManagement/timers_module.c]: Is this conversation to ns correct?

File: 06-TimeManagement/timers_module.c [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+	} else {
+		fs = get_fs();
+		set_fs(get_ds());
+		f->f_op->read(f, buf_file, 64, &f->f_pos);
+		set_fs(fs);
+	}
+	filp_close(f, NULL);
+
+	buf_file_sub = strsep(&buf_file_sub, " ");
+	memset(module_data->buf_str, 0, 32);
```
Comment (yekovalyov) [inline 06-TimeManagement/timers_module.c]: Is this memset operation required?

**Comment by yekovalyov** [comment]
Did You fix all warning from checkpatch.pl?

**Comment by BekirovBekir** [comment]
I've fixed the all warnings except one warning is related to return condition into if-else operator. It really is necessary to fix this warning? Thanks.

**Pull request merged**